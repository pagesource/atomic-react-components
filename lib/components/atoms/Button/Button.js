// @flow
import React from 'react';
import type { Node, ComponentType } from 'react';
import styled from 'styled-components';

import { getLabelFromChildren } from '../../../core/utils/elementLocator';
import styles from './Button.style';

type Props = {
  className?: string,
  children: Node,
  inheritedStyles?: string,
  type?: string,
  primary?: boolean,
  secondary?: boolean,
  tertiary?: boolean,
  disabled?: boolean,
  variation?: string,
  transparent?: boolean,
  ariaLabel?: string,
  elementLocator?: string,
};

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  tertiary,
  variation,
  disabled,
  transparent,
  ariaLabel,
  elementLocator,
  ...others
}: Props): Node => (
  <button
    {...others}
    aria-label={ariaLabel || null}
    className={className}
    disabled={disabled}
    type={type}
    data-locator={
      elementLocator || (type ? `button-${type}-${getLabelFromChildren(children)}` : '')
    }
    variation={variation}
  >
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  tertiary: false,
  disabled: false,
  variation: '',
  transparent: false,
  ariaLabel: '',
  className: '',
  elementLocator: '',
};

const StyledButton: ComponentType<Props> = styled(Button)`
  ${styles};
`;
export default StyledButton;

export { Button as ButtonVanilla };
