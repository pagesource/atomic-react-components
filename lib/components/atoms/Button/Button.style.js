import { css } from 'styled-components';

const ButtonStyles = css`
  border: none;
  border-radius: 0;
  position: relative;
  min-height: 48px;
  cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
  ${(props) => `
      ${props.theme.colors.CTA_PRIMARY};
      ${props.theme.typography.FONT_CTA};
      padding: 0 ${props.theme.spacing.UNIT_2};
  `};
  ${(props) => props.secondary && `${props.theme.colors.CTA_SECONDARY}`};
  ${(props) =>
    props.tertiary
      ? `
    ${props.theme.colors.CTA_TERTIARY};
    ${props.theme.border.CTA_TERTIARY};
  `
      : ``};
  ${(props) =>
    props.variation === 'icon-only' &&
    `
    padding: 0 ${props.theme.spacing.UNIT_1};
    ${props.theme.colors.CTA_ICON}
  `}
  ${(props) => props.fullWidth && ` width: 100% ;`};
  ${(props) =>
    props.variation === 'bullets' &&
    `
      border-radius: 50%;
      text-indent: -1000px;
      overflow: hidden;
      width: 10px;
      height: 10px;
      padding: 0;
      min-height: initial;
      margin: ${props.theme.spacing.UNIT_1};
  }`}

  ${(props) =>
    props.transparent &&
    `
    ${props.theme.typography.FONT_CTA};
    ${props.theme.colors.CTA_TRANSPARENT}
  `};

  ${(props) => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default ButtonStyles;
