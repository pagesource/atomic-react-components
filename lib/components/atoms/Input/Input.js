// @flow
import React from 'react';
import type { ComponentType } from 'react';
import styled from 'styled-components';
import styles from './Input.style';

type Props = {
  id: string,
  className?: string,
  type?: string,
  name: string,
  value?: string | number,
  placeholder?: string,
  ariaLabel?: string,
  elementLocator?: string,
  ref?: ?HTMLInputElement,
};

const Input = React.forwardRef<Props, HTMLInputElement>(
  (
    { id, className, type, name, value, placeholder, ariaLabel, elementLocator, ...others }: Props,
    ref
  ) => {
    const propArray = {
      id,
      ref,
      'aria-label': ariaLabel,
      className,
      name,
      type,
      placeholder,
      'data-locator': elementLocator || `input-${type}-${name}`,
      ...others,
    };
    if (type === 'radio') propArray.value = value;
    return React.createElement('input', propArray);
  }
);

// $flow-disable-line  eslint-disable-line
Input.defaultProps = {
  placeholder: '',
  ariaLabel: '',
  className: '',
  type: 'text',
  value: '',
  elementLocator: '',
  ref: null,
};

const StyledInput: ComponentType<Props> = styled(Input)`
  ${styles};
`;

export default StyledInput;
export { Input as InputVanilla };
