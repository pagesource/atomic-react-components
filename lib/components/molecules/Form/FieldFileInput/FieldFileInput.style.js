import { css } from 'styled-components';

// TODO: Fix the duplicate styling in input.style.js
const styles = css`
  margin-bottom: ${(props) => props.theme.spacing.UNIT_2};
  ${(props) => `${props.theme.border.DRAG_FILE_FIELD}`};
  display: flex;
  justify-content: space-between;
  padding: ${(props) => props.theme.spacing.UNIT_2};
  input {
    display: none;
  }
  label {
    display: none;
  }
`;

export default styles;
