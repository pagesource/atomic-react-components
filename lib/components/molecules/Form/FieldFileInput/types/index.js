// @flow

export type FieldFileInputProps = {
  id: string,
  className?: string,
  register: Function,
  setValue: Function,
  name: string,
  label: Node,
  labelProps?: Object,
  fieldProps?: Object,
  localeText: {
    btnTextPrimary: string,
    btnTextSecondary: string,
    dragHelperText: string,
  },
  onChange?: Function,
  onBlur?: Function,
};
