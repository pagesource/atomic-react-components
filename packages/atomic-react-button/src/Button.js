import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getLabelFromChildren } from '@xt-pagesource/atomic-react-utils';
import styles from './Button.style';

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  tertiary,
  variation,
  disabled,
  transparent,
  ariaLabel,
  elementLocator,
  ...others
}) => (
  <button
    {...others}
    aria-label={ariaLabel || null}
    className={className}
    disabled={disabled}
    type={type}
    data-locator={
      elementLocator || (type ? `button-${type}-${getLabelFromChildren(children)}` : '')
    }
    variation={variation}
  >
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  tertiary: false,
  disabled: false,
  variation: '',
  transparent: false,
  ariaLabel: '',
  className: '',
  elementLocator: '',
};

Button.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  inheritedStyles: PropTypes.string,
  type: PropTypes.string,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  tertiary: PropTypes.bool,
  disabled: PropTypes.bool,
  variation: PropTypes.string,
  transparent: PropTypes.bool,
  ariaLabel: PropTypes.string,
  elementLocator: PropTypes.string,
};

const StyledButton = styled(Button)`
  ${styles};
`;
export default StyledButton;

export { Button as ButtonVanilla };
