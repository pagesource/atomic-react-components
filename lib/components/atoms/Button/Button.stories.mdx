import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { primaryButton, primaryButtonDisabled, secondaryButton, secondaryButtonDisabled, customButtonLocator} from './Button.mock';
import Button, { ButtonVanilla } from '.';

<Meta title="Atoms/Buttons" component={Button} />

### Vanila  Button

<Preview>
  <Story name="Vanila Button">
   <ButtonVanilla {...primaryButton}>Submit</ButtonVanilla>
  </Story>
   <Story name="Vanila Disabled Button">
   <ButtonVanilla {...primaryButtonDisabled}>Submit</ButtonVanilla>
  </Story>
</Preview>


### Primary  Button

<Preview>
  <Story name="Primary Styled Button">
   <Button {...primaryButton}>Submit</Button>
  </Story>
   <Story name="Primary Styled Disabled Button">
   <Button {...primaryButtonDisabled}>Submit</Button>
  </Story>
</Preview>


### Secondary Button

<Preview>
  <Story name="Secondary Styled Button">
   <Button {...secondaryButton}>Submit</Button>
  </Story>
   <Story name="Secondary Styled Disabled Button">
   <Button {...secondaryButtonDisabled}>Submit</Button>
  </Story>
</Preview>

### custom Button Locator

<Preview>
  <Story name="Custom  Button">
   <Button {...customButtonLocator}>Submit</Button>
  </Story>
</Preview>

## Props ðŸ”§

<Props of={ButtonVanilla} />
