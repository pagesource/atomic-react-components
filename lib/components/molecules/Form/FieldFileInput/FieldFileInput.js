// @flow
import React, { useEffect } from 'react';
import type { Node, ComponentType } from 'react';
import styled from 'styled-components';

import Input from '../../../atoms/Input';
import Label from '../../../atoms/Label';
import Button from '../../../atoms/Button';
import styles from './FieldFileInput.style';
import type { FieldFileInputProps } from './types';
import useOpenFileDialog from '../../../../hooks/useOpenFileDialog';

const FieldFileInput = ({
  id,
  className,
  name,
  label,
  onChange,
  onBlur,
  labelProps,
  fieldProps,
  register,
  setValue,
  localeText,
  ...others
}: FieldFileInputProps): Node => {
  const fileInput = React.createRef();
  const { openFileDialog } = useOpenFileDialog();

  useEffect(() => {
    register({ name });
  }, [register]);

  const handleChange = (e) => {
    setValue(name, e.target.files); // you get all the files object here
  };

  return (
    <div className={className}>
      <Label htmlFor={id} {...labelProps}>
        {label}
      </Label>
      <Input
        {...others}
        id={id}
        name={name}
        type="file"
        ref={fileInput}
        {...fieldProps}
        onChange={handleChange}
      />
      <Button tertiary type="button" onClick={() => openFileDialog(fileInput)}>
        {localeText.btnTextPrimary}
      </Button>
      <Button primary type="button">
        {localeText.btnTextSecondary}
      </Button>
    </div>
  );
};

FieldFileInput.defaultProps = {
  className: 'fieldFileInput',
  labelProps: {},
  fieldProps: {},
  onChange: null,
  onBlur: null,
  localeText: {
    btnTextPrimary: 'Choose File(s)',
    btnTextSecondary: 'Upload...',
  },
};

const StyledFieldFileInput: ComponentType<FieldFileInputProps> = styled(FieldFileInput)`
  ${styles};
`;

export default StyledFieldFileInput;

export { FieldFileInput as FieldFileInputVanilla };
