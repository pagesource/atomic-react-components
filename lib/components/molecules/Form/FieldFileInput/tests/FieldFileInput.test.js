import React from 'react';

import { shallowWithTheme } from '../../../../../../__mock__/themeMock';
import { FieldFileInputVanilla } from '../FieldFileInput';
import Button from '../../../../atoms/Button';
import Input from '../../../../atoms/Input';

const defaultProps = {
  name: 'fileupload',
  localeText: {
    btnTextPrimary: 'Choose File(s)',
    btnTextSecondary: 'Upload...',
    dragHelperText: 'Drop Files Here...',
  },
};

describe('FieldInput Component', () => {
  let FieldInput = '';
  beforeEach(() => {
    FieldInput = shallowWithTheme(<FieldFileInputVanilla {...defaultProps} />);
  });

  test('should render correctly', () => {
    expect(FieldInput).toMatchSnapshot();
  });

  test.skip('should render button with type = "button".', () => {
    expect(FieldInput.find('Button')).toHaveLength(2);
  });

  test('should render 1st button with theme prop type, tertiary', () => {
    // FieldInput = renderWithTheme(<FieldFileInputVanilla {...defaultProps} />);
    const doesExist = FieldInput.find(Button).first().prop('tertiary');
    expect(doesExist).toBeTruthy();
  });

  test('should render 2nd button with theme prop type, primary', () => {
    const doesExist = FieldInput.find(Button).at(1).prop('primary');
    expect(doesExist).toBeTruthy();
  });

  test('should match the button text rendered', () => {
    const button = FieldInput.find(Button).first();
    expect(button.text()).toContain('Choose');
  });

  test('should match the button text rendered for second button', () => {
    const button = FieldInput.find(Button).at(1);
    expect(button.text()).toContain('Upload...');
  });

  test('should match the label rendered correctly.', () => {
    const doesExist = FieldInput.find('label');
    expect(doesExist).toBeTruthy();
  });

  test('should match the input rendered correctly.', () => {
    const doesExist = FieldInput.find('input');
    expect(doesExist).toBeTruthy();
  });

  test.skip('should match the input rendered correctly on change event simulation.', () => {
    const inputElement = FieldInput.find('Input');
    const wrapper = inputElement.simulate('change');
    expect(wrapper.find('Input').length).toBe(1);
  });

  test('should match the input rendered correctly on blur event simulation.', () => {
    const inputElement = FieldInput.find(Input);
    const wrapper = inputElement.simulate('blur');
    expect(wrapper.find(Input).length).toBe(1);
  });
});
