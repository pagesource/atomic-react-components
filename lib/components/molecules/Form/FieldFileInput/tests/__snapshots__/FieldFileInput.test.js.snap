// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput Component should render correctly 1`] = `
<div
  className="fieldFileInput"
>
  <Label
    className=""
    elementLocator=""
  />
  <Input__StyledInput
    name="fileupload"
    onChange={[Function]}
    type="file"
  />
  <Button__StyledButton
    onClick={[Function]}
    tertiary={true}
    type="button"
  >
    Choose File(s)
  </Button__StyledButton>
  <Button__StyledButton
    primary={true}
    type="button"
  >
    Upload...
  </Button__StyledButton>
</div>
`;
